aform/src/components/form/ADropdown.vue

```diff
@@ -29,86 +28,89 @@
 </template>
 
 <script setup lang="ts">
-import { onMounted, onUnmounted, ref /* useTemplateRef */ } from 'vue'
-
-const { label, items, isAsync } = defineProps<{
+import { ref, useTemplateRef } from 'vue'
+import { vOnClickOutside } from '@vueuse/components'
+const { label, items, isAsync, filterFunction } = defineProps<{
 	label: string
 	items?: string[]
 	isAsync?: boolean
+	filterFunction?: (search: string) => Promise<string[]>
 }>()
-
-const emit = defineEmits(['filterChanged'])
-
+const autocomplete = useTemplateRef<HTMLElement>('autocomplete')
 const results = ref(items)
 const search = defineModel<string>()
 const isLoading = ref(false)
 const arrowCounter = ref(0)
 const isOpen = ref(false)
-// const mopInput = useTemplateRef<HTMLInputElement>('mopInput')
-
-onMounted(() => {
-	document.addEventListener('click', handleClickOutside)
-	filterResults()
-})
-
-onUnmounted(() => {
-	document.removeEventListener('click', handleClickOutside)
-})
 
-const setResult = result => {
-	search.value = result
+const closeResultsHandler = () => {
 	closeResults()
+	arrowCounter.value = 0
```

Comment:
Should this be unset (`null` or `undefined`?) if there's no active item being selected anymore?